myriad-scheduler/src/main/java/org/apache/myriad/scheduler/fgs/YarnNodeCapacityManager.java

```diff
@@ -135,9 +136,9 @@ private void removeYarnTask(RMContainer rmContainer) {
        */
       state.makeTaskKillable(taskId);
       
-      Node node = retrieveNode(rmContainer);
-      if (node != null) {
-        RMNode rmNode = node.getNode().getRMNode();
+      Optional<Node> optNode = retrieveNode(rmContainer);
+      if (optNode.isPresent()) {
+        RMNode rmNode = optNode.get().getNode().getRMNode();
```

Comment:
Ah, gotcha, I will update accordingly.

--John

On Wed, Aug 24, 2016 at 11:53 AM, Darin <notifications@github.com> wrote:

> In myriad-scheduler/src/main/java/org/apache/myriad/scheduler/fgs/
> YarnNodeCapacityManager.java
> <https://github.com/apache/incubator-myriad/pull/91#discussion_r76084477>:
>
> > @@ -135,9 +136,9 @@ private void removeYarnTask(RMContainer rmContainer) {
> >         */
> >        state.makeTaskKillable(taskId);
> >
> > -      Node node = retrieveNode(rmContainer);
> > -      if (node != null) {
> > -        RMNode rmNode = node.getNode().getRMNode();
> > +      Optional<Node> optNode = retrieveNode(rmContainer);
> > +      if (optNode.isPresent()) {
> > +        RMNode rmNode = optNode.get().getNode().getRMNode();
>
> While I'm a fan of optionals this usage is a bit overkill, in fact I think
> it makes this less clear at the cost of extract object creation.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-myriad/pull/91/files/edf6bae875bc016bb50fb3b0517c320d6af5c015#r76084477>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKSRgUewUJetf8wyogVvGgJmw9DPMEVnks5qjGjigaJpZM4JqHXc>
> .
>